### ISSUE TITLE
Just Feedback Really

### ISSUE THREAD

**Issue by WHumphreys** [fireeye/commando-vm#42]
I quite like this but for what its worth this is my feedback.

Setup on Windows 10. No install issues on the whole. I did notice a few script errors by chance (as I was doing other stuff, its a long install) that I didnt read properly but saying that everything seems to work fine.

The README.txt on the desktop after installation:

  * requests spelt wrong "reqeuests"
  * Github URL is wrong "https://github.com/fireeye/commandovm" 
    should be "https://github.com/fireeye/commando-vm"

Visual Studio 2019 is out now so why install 2017. .Net Core really needs 2019 and it is the way forward, especially for cross platform.

In note of that why not add some sort of config file that lets you pick versions of everything.

Maybe I actually own "paid" for versions of software so I dont want the free versions. An extra config file I could change before running the installer would help. (I could change the installer but it gets a bit messy if you change stuff).

I think Ghidra "https://github.com/NationalSecurityAgency/ghidra" should be included. In fact I think "Hacking" and "Reversing" Installs should be on the same machine these days as they sort of go hand in hand now. Why not mix your other VM (that name escapes me this moment in time). And just have one.

Though a "little personal and you would expect somebody installing this to know how beforehand" but the VM recommended disk size is smaller than some of my password text files... VM's dont have to use the full size on install so I think it should be made much larger (with a note maybe as why relative to its install size"

I'm also surprised stuff like Wpscan, Nikto, Nessus and various other very common (but quick useful) scanning tools where not included.

On the whole though so far I like it.

**Comment by htnhan** [comment]
@WHumphreys 

Thank you for using and supporting Commando VM. 

I have created #4 to track the issue with incorrect URL in readme.txt

The long term plan for Commando VM is, like you said, you merge with FLARE VM and allow a custom configuration for every user. However, that is way down the road. 

Ghidra is strictly a reverse engineering tool. We include Ghidra in FLARE VM. However, within the context of red team engagement, I am not sure how often Ghidra, or IDA, comes in handy. Therefore, we do not include Ghridra at the moment.

**Comment by day1player** [comment]
@WHumphreys Thank you very much for the detailed feedback!

We will be expanding the tool set to include as much as we can based on user feedback. Of all of the tools you mentioned, Nessus might pose the only problem as far as easy install goes. I did look into this a few months ago, and Tenable generates some dynamic URLs and makes you agree to some terms before giving you the dynamic link, so it's not something that can be easily turned into a chocolatey package. It would be great to have though, I agree.

As far as Visual Studio is concerned I will look into 2019, I agree we need to keep tooling up-to-date with industry but this is a fairly large change especially as far as package management goes. I think we can add this to the roadmap and have it in one of the next version releases.

Let us know what other feedback you have as you use it! Really glad you like it so far, thanks for taking the time to get on and create this issue.

**Comment by WHumphreys** [comment]
Although this is just my own personal opinion as it doesn't really have a right answer. But I never like the "script" that tries to be what I like to call a "forklifttruckbuscar".

Everybody has basic needs but very different objectives. If I was doing this I would have an extra layer over  the "install" mechanisms.

Maybe like an initial script that gets the ball rolling with some initial setup but then what is basically a simple package manager that simple sits on "top".

This has the advantage that I can actually choose what I want to install / remove / update and something that would also be great is good description of what said software is.  One of my gripes with Kali is having lots of command line tools that I wish I had known about that was there all along.

Then its easy to have many versions of software like "Visual Studio" so you can pick one. It means I can ignore free versions I may have full versions of, it may even include full versions but warn of requiring keys etc.

Being as I would assume that folk using this are pretty IT literate I think automating to much takes away to much control and assumes to much.

**Comment by day1player** [comment]
@WHumphreys We are looking to release some granular package management features in August. We submitted the tool to Black Hat USA and hope to release CVM 2.0 at that time, which will include Kali Linux on WSL, and the ability for users to create a custom package file that determines what tools to install for the user. 

As for Visual Studio when deciding which version to install (pro or community) I found out that upgrading to the professional version was almost as simple as entering a product key on the community version, so decided to go with community. However, in the 2.0 release, that should be a non issue because we can create a "pro" package for people to install as well. 

If you have anymore feedback just keep it coming!

Thanks again.
